src/com/rcv/CVRReader.java

```diff
@@ -103,21 +105,36 @@ private static Sheet getFirstSheet(String excelFilePath) {
       String cvrFileName = new File(excelFilePath).getName();
       // cvrIndex for generating cvrIDs
       int cvrIndex = 1;
+      // counts of unrecognized candidate strings
+      Map<String, Integer> unrecognizedCandidateCounts = new HashMap<>();
 
       // Iterate through all rows and create a CastVoteRecord for each row
       while (iterator.hasNext()) {
         // cvr is the object parsed from the row
-        CastVoteRecord cvr = parseRow(iterator.next(), cvrFileName, cvrIndex++);
-        castVoteRecords.add(cvr);
+        try {
+          CastVoteRecord cvr = parseRow(iterator.next(), cvrFileName, cvrIndex++);
+          castVoteRecords.add(cvr);
+        } catch (UnrecognizedCandidateException e) {
+          // increment the count
+          unrecognizedCandidateCounts.merge(e.getName(), 1, Integer::sum);
+        }
+      }
+
+      if (unrecognizedCandidateCounts.size() > 0) {
+        throw new SourceWithUnrecognizedCandidatesException(unrecognizedCandidateCounts);
       }
     }
+
     return castVoteRecords;
   }
 
   // function: parseRow
   // purpose: parse a single row into a CastVoteRecord
   // returns: a CastVoteRecord object
-  private CastVoteRecord parseRow(Row castVoteRecordRow, String cvrFileName, int cvrIndex) {
+  private CastVoteRecord parseRow(
+      Row castVoteRecordRow,
+      String cvrFileName,
+      int cvrIndex) throws UnrecognizedCandidateException {
```

Comment:
I know this looks nicer, but my google-java-format plugin is fighting this when I do a Reformat Code. Do you have it enabled on your end?

Reply:
I thought I had enabled everything, but clearly it's not working. Let me try again...